examples/docs_snippets/docs_snippets/guides/dagster/local_to_prod_tutorial/resources/resources_v2.py

```diff
@@ -0,0 +1,41 @@
+from typing import Any, Dict, Optional
+
+from dagster import resource
+
+# start_mock
+# resources.py
+
+
+class StubHNClient:
+    """
+    Hacker News Client that returns fake data
+    """
+
+    def __init__(self):
+        self.data = {
+            1: {
+                "id": 1,
+                "type": "comment",
+                "title": "the first comment",
+                "by": "user1",
+            },
+            2: {"id": 2, "type": "story", "title": "an awesome story", "by": "user2"},
+        }
+
+    def fetch_item_by_id(self, item_id: int) -> Optional[Dict[str, Any]]:
+        return self.data.get(item_id)
+
+    def fetch_max_item_id(self) -> int:
+        return 2
+
+    @property
+    def item_field_names(self):
```

Comment:
I think we want this property to be available on `hn_api_client` too, passing in an empty list causes a create table failure error

Reply:
so I purposely removed it from the guide because the list is really long and black autoformats it to one entry per line and that causes the code snippet to be huge. There's a note about it after the rewrite of the `items` asset but maybe it should be harder to miss. what do you think?

In the real code example (in the examples directory) the property is fully defined

Reply:
Got it, that makes sense. I think it's fine since the property is defined in the examples file